textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -0,0 +1,166 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IDictionary;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.field.IntegerField;
+import edu.uci.ics.textdb.common.field.StringField;
+
+/**
+ * @author Sudeep [inkudo]
+ * 
+ */
+public class DictionaryMatcher implements IOperator {
+
+    private IOperator operator;
+    private IDictionary dictionary;
+    private String dictionaryValue;
+    private int positionIndex; // next position in the field to be checked.
+    private int fieldIndex; // Index of the next field to be checked.
+    private int spanIndexValue; // Starting position of the matched dictionary
+                                // string
+    private String fieldName;
+    private ITuple dataTuple;
+    private List<IField> fields;
+    private List<Attribute> schema;
+
+    public DictionaryMatcher(IDictionary dictionary, IOperator operator) {
+        this.operator = operator;
+        this.dictionary = dictionary;
+    }
+
+    /**
+     * @about Opens dictionary matcher. Initializes positionIndex and fieldIndex
+     *        Gets first sourceoperator tuple and dictionary value.
+     */
+    @Override
+    public void open() throws Exception {
+        try {
+            positionIndex = 0;
+            fieldIndex = 0;
+            operator.open();
+            dictionaryValue = dictionary.getNextValue();
+            dataTuple = operator.getNextTuple();
+            fields = dataTuple.getFields();
+            schema = dataTuple.getSchema();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @about Gets next matched tuple. Returns a new span tuple including the
+     *        span results. Performs a scan based search, gets the dictionary
+     *        value and scans all the documents for matches
+     * 
+     * @overview Loop through the dictionary entries. For each dictionary entry,
+     *           loop through the tuples in the operator. For each tuple, loop
+     *           through all the fields. For each field, loop through all the
+     *           matches.
+     */
+    @Override
+    public ITuple getNextTuple() throws Exception {
+        if (fieldIndex < fields.size()) {
+            IField dataField = dataTuple.getField(fieldIndex);
+            if (dataField instanceof StringField) {
+                String fieldValue = ((StringField) dataField).getValue();
+                fieldValue = fieldValue.toLowerCase();
+                dictionaryValue = dictionaryValue.toLowerCase();
```

Comment:
Sure prof. I thought of committing that change along with the changes for
the token boundaries. In the meantime, we have already merged the PR. I'll
merge these changes in the next PR.
ᐧ

On Tue, Apr 19, 2016 at 11:05 AM, Chen Li <notifications@github.com> wrote:

> In
> textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java
> <https://github.com/TextDB/textdb/pull/61#discussion_r60280632>:
>
> > +     *        span results. Performs a scan based search, gets the dictionary
> > +     *        value and scans all the documents for matches
> > +     *
> > +     * @overview Loop through the dictionary entries. For each dictionary entry,
> > +     *           loop through the tuples in the operator. For each tuple, loop
> > +     *           through all the fields. For each field, loop through all the
> > +     *           matches.
> > +     */
> > +    @Override
> > +    public ITuple getNextTuple() throws Exception {
> > +        if (fieldIndex < fields.size()) {
> > +            IField dataField = dataTuple.getField(fieldIndex);
> > +            if (dataField instanceof StringField) {
> > +                String fieldValue = ((StringField) dataField).getValue();
> > +                fieldValue = fieldValue.toLowerCase();
> > +                dictionaryValue = dictionaryValue.toLowerCase();
>
> @sandeepreddy602 <https://github.com/sandeepreddy602> and team 1: please
> make sure to take care of this comment (even though this PR is merged).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/TextDB/textdb/pull/61/files/d5794b1042119c0eea83bc3f34d3263e6ab6478b#r60280632>
>